src/App.tsx

```diff
@@ -1,60 +1,153 @@
-import React from 'react';
+import {
+  FC,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
-import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
+import cn from 'classnames';
+import { PostsList } from './components/PostsList/PostsList';
+import { PostDetails } from './components/PostDetails/PostDetails';
+import { UserSelector } from './components/UserSelector/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './api/users';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getPosts } from './api/posts';
+
+export const App: FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [noPosts, setNoPosts] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const fetchUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch {
+      setIsError(true);
+    }
+  };
+
+  const fetchUserPosts = async () => {
+    setNoPosts(false);
+    setIsLoading(true);
+    setIsError(false);
+
+    try {
+      const postsFromServer = await getPosts(selectedUserId);
+
+      if (postsFromServer.length === 0) {
+        setNoPosts(true);
+      }
+
+      setPosts(postsFromServer);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleSelectUser = useCallback((user: User) => {
+    if (user.id !== selectedUserId) {
+      setSelectedUserId(user.id);
+      setSelectedPostId(0);
+      setPosts([]);
+    }
+  }, [selectedUserId]);
+
+  const selectedPost = useMemo(() => {
+    return posts.find(({ id }) => id === selectedPostId);
+  }, [selectedPostId]);
+
+  useEffect(() => {
+    if (selectedUserId) {
```

Comment:
You have id as a number so the user with id 0 will never be selected? I would use null as initial value for selectedUserId and here would change the condition for selectedUserId not to be the null